src/sources/docker_logs.rs

```diff
@@ -75,26 +77,39 @@ impl Default for DockerLogsConfig {
 }
 
 impl DockerLogsConfig {
-    fn container_name_included<'a>(
+    fn container_name_or_id_included<'a>(
         &self,
         id: &str,
         names: impl IntoIterator<Item = &'a str>,
     ) -> bool {
-        if let Some(include_containers) = &self.include_containers {
-            let id_flag = include_containers
-                .iter()
-                .any(|include| id.starts_with(include));
+        let containers: Vec<String> = names.into_iter().map(|i| i.into()).collect();
 
-            let name_flag = names.into_iter().any(|name| {
-                include_containers
-                    .iter()
-                    .any(|include| name.starts_with(include))
-            });
+        // Containers are included by default
+        let mut included: bool = true;
 
-            id_flag || name_flag
-        } else {
-            true
+        // Container remains included if include_containers is specified
+        // and container name or ID matches the inclusion list
+        if let Some(include_list) = &self.include_containers {
+            included = Self::name_or_id_matches(id, &containers, include_list);
+        }
+
+        // Container gets excluded if exclude_containers is specified
+        // and container name or ID matches the exclusion list
+        if let Some(exclude_list) = &self.exclude_containers {
```

Comment:
Not a big deal, but I think you could avoid even checking exclude containers for a given container if `included` is `false` at this point.

An alternative way to write this might be:

```rust
match (&self.include_containers, &self.exclude_containers) {
  (Some(include_list), Some(exclude_list)) => Self::name_or_id_matches(id, &containers, include_list) && !(Self::name_or_id_matches(id, &containers, exclude_list)),
  (Some(include_list), None) => Self::name_or_id_matches(id, &containers, include_list),
  (None, Some(exclude_list)) => !Self::name_or_id_matches(id, &containers, exclude_list),
  (None, None) => true,
}
```

Reply:
Oh man, somehow I always forget just how powerful match statements are. Great suggestion.